src/main/java/de/tum/in/www1/artemis/web/rest/CompetencyResource.java

```diff
@@ -276,6 +271,34 @@ public ResponseEntity<Competency> importCompetency(@PathVariable long courseId,
         return ResponseEntity.created(new URI("/api/courses/" + courseId + "/competencies/" + competencyToImport.getId())).body(competencyToImport);
     }
 
+    /**
+     * Imports all competencies of the source course (and optionally their relations) into another
+     *
+     * @param courseId        the id of the course to import into
+     * @param sourceCourseId  the id of the course to import from
+     * @param importRelations if relations should be imported aswell
+     * @return the ResponseEntity with status 201 (Created) and with body containing the imported competencies (and relations)
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/courses/{courseId}/competencies/import-all/{sourceCourseId}")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<List<CompetencyWithTailRelationDTO>> importAllCompetenciesFromCourse(@PathVariable long courseId, @PathVariable long sourceCourseId,
+            @RequestParam(defaultValue = "false") boolean importRelations) throws URISyntaxException {
+        log.info("REST request to all competencies from course {} into course {}", sourceCourseId, courseId);
+
+        if (courseId == sourceCourseId) {
+            throw new ConflictException("Cannot import from a course into itself", "Course", "courseCycle");
```

Comment:
You should also add translations in the client for the `courseCycle` key.

Reply:
Was discussed and left out because the message should never get displayed to the user.